Analysis/DEEP2_R23_O32/Binning/N_Bins_Grid_Analysis.py

```diff
@@ -162,21 +191,29 @@ def n_times_binned(fitspath, dataset, pdf_pages, outfile, n_split, R23, O32, SNR
         y_average = [yBar[jj], yBar[jj]]
         plt.plot(x_value, y_value, linewidth=0.3, color='b')
         plt.plot(x_value, y_average, linewidth=0.3, color='g')
+
     fig.savefig(pdf_pages, format='pdf')
     pdf_pages.close()
 
-    np.savez(outfile, locator=locator, R23_minimum=R23_minimum, O32_minimum=O32_minimum, Number_inbin=Number_inbin)
-
-    n1 = ('ID', 'R23_minimum', 'O32_minimum', 'xBar', 'yBar', 'R23_median', 'O32_median', 'area')
-    tab1 = Table([n_bins_range, R23_lowlimit, O32_lowlimit, xBar, yBar, R23_medians, O32_medians, area], names=n1)
-    asc.write(tab1, fitspath+'/'+dataset+'_binning_averages.tbl', format='fixed_width_two_line')
+    np.savez(outfile, locator=locator, R23_minimum=R23_minimum, O32_minimum=O32_minimum, Number_inbin=number_inbin)
     
-    fig.clear()
-
-    n2 = ('R23', 'O32', 'SN_5007', 'Bin_number')
-    tab2 = Table([R23, O32, SNR3, Bin_number], names=n2)
-    asc.write(tab2, fitspath+'/' + dataset+'_2d_binning_datadet3.tbl', format='fixed_width_two_line')
+    n1 = ('bin_ID', 'N_stack', 'logR23_min', 'logO32_min', 'logR23_avg', 'logO32_avg',
+          'logR23_median', 'logO32_median', 'logR23_max', 'logO32_max')
+    tab1 = Table([n_bins_range, area, np.log10(R23_lowlimit), np.log10(O32_lowlimit), np.log10(xBar),
+                  np.log10(yBar), np.log10(R23_medians), np.log10(O32_medians), np.log10(R23_maxval),
+                  np.log10(O32_maxval)], names=n1)
+    # used to be called +dataset+'_binning_averages.tbl
+    asc.write(tab1, fitspath+'/bin_info.tbl', format='fixed_width_two_line')
```

Comment:
Also, this isn't using the `msc` variable inputs for filename, `filename_dict`

A `join` might be desired`.

Reply:
done